compiler/rustc_codegen_llvm/src/debuginfo/metadata.rs

```diff
@@ -922,6 +922,27 @@ fn build_field_di_node<'ll, 'tcx>(
     }
 }
 
+/// Returns the `DIFlags` corresponding to the visibility of the item identified by `did`.
+///
+/// `DIFlags::Flag{Public,Protected,Private}` correspond to `DW_AT_accessibility`
+/// (public/protected/private) aren't exactly right for Rust, but neither is `DW_AT_visibility`
+/// (local/exported/qualified), and there's no way to set `DW_AT_visibility` in LLVM's API.
+fn visibility_di_flags<'ll, 'tcx>(
+    cx: &CodegenCx<'ll, 'tcx>,
+    did: DefId,
+    type_did: DefId,
+) -> DIFlags {
+    let parent_did = cx.tcx.parent(type_did);
+    let visibility = cx.tcx.visibility(did);
```

Comment:
I was going to comment that it might make more sense to calculate the visibility from the perspective of the crate root (or, rather, as if you were using the crate as a dependency?) but I don't see why that necessarily be better and the current implementation is simpler and matches the visibility of the declaration site (even if the *effective* visibility ends up being more hidden due to the visibility/exports of parent modules). 

Just noting that is another possible way to implement this but I don't see a strong reason to do that at this time.